tests/specs/functional/lemmas-spec.k

```diff
@@ -256,4 +260,18 @@ module LEMMAS-SPEC
     claim <k> runLemma ( N &Int maxUInt8   ) => doneLemma ( N ) ... </k> requires #rangeUInt(8,   N)
     claim <k> runLemma ( maxUInt8 &Int N   ) => doneLemma ( N ) ... </k> requires #rangeUInt(8,   N)
 
+ // ecrecover
+ // ---------
+    claim [ecrec-length]: <k> runLemma  ( lengthBytes ( #ecrec ( _ , _ , _ , _ ) ) <=Int 32 )
+                           => doneLemma ( true ) ... </k>
+
+    claim [ecrec-split]: <k> runLemma  ( #range ( #buf(160, 0) +Bytes _BUF , lengthBytes ( #ecrec ( HASH , SIGV , SIGR , SIGS ) ) +Int 128 , 32 -Int lengthBytes ( #ecrec ( HASH , SIGV , SIGR , SIGS ) ) ) )
+                          => doneLemma ( #range ( #buf( 32, 0)             , lengthBytes ( #ecrec ( HASH , SIGV , SIGR , SIGS ) )          , 32 -Int lengthBytes ( #ecrec ( HASH , SIGV , SIGR , SIGS ) ) ) ) ... </k>
```

Comment:
I don't quite understand this claim I don't think. It's basically saying, either:

- `#ecrec` is valid (returns an address), and so the overall read of `#range` will return an empty bytes, or
- `#ecrec` is _invalid_ (returns empty), and so the overall read of `#range` will return 32 zeros?

Am I reading this right? Why does such a term show up?

Reply:
This appears as part of a larger expression:
```
#ecrec(...) +Bytes #range(#buf(32, 0), lengthBytes(#ecrec(...)), 32 -Int lengthBytes(#ecrec(...)))
```
It seems that this expression is essentially doing `padRightBytes(#ecrec(...), 32, 0)`:
- If `#ecrec` returns 0 bytes, then it will be appended with 32 zeros.
- If `#ecrec` returns 32 bytes, then it will be appended with an empty bytes.

I can add the `#ecrec(...) +Bytes ...` part to the claim to make its context more clear, if you think that would be better.

Reply:
No, it's fine, thanks for the explanation!